File: 卡码/代码随想录/Python/二叉树/02-二叉树属性/LeetCode_0257.py
#257. 二叉树的所有路径
from typing import *
from ..TreeNode import *

class Solution:
  def traversal(self,root:Optional[TreeNode],path:str,res:List[str]):
    path+=str(root.val)
    #找到叶子节点
    if not root.left and not root.right:
      res.append(path)
      return
    if root.left:
      self.traversal(root.left,path+"->",res)
    if root.right:
      self.traversal(root.right,path+"->",res)

  def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
    res:List[str]=[]
    path:str=""
    if not root:
      return res
    self.traversal(root,path,res)
    return res

File: 卡码/代码随想录/JS/二叉树/02-二叉树属性/LeetCode_0257.js
//257. 二叉树的所有路径
import TreeNode from "../TreeNode.js";

const traversal=(root,path,res)=>{
  path+=root.val;
  //找到叶子节点
  if(!root.left&&!root.right){
    res.push(path);
    return;
  }
  if(root.left) traversal(root.left,path+"->",res);
  if(root.right) traversal(root.right,path+"->",res);
}
const binaryTreePaths = function(root) {
  const res=new Array();
  let path="";
  if(!root) return res;
  traversal(root,path,res);
  return res;
};